File: recipe-go/ingredient.go
package main



type Ingredient struct
{
    Name string
    Unit uint64
    Quantity uint64
    Description string
    id uint64 // private
}

  func NewIngredient(name string, unit uint64, quantity uint64, description string, id uint64) (*Ingredient, error) {
    return &Ingredient {
        Name : name,
        Unit : unit,
        Quantity : quantity,
        Description : description,
        id : id,
    }, nil
}




func (ingredient Ingredient) Id() uint64 {
  return ingredient.id
}

func (ingredient * Ingredient) SetId(uint64 id) uint64 {
  ingredient.id = id
}

File: recipe-rust/src/ingredient.rs
pub struct Ingredient {
    pub name: String,
    pub unit: u64,
    pub quantity: u64,
    pub description: String,
    id: u64,
}
impl Ingredient {
    // Constructor to create a new Ingredient
    pub fn new(name: String, unit: u64, quantity: u64, description: String, id: u64) -> Self {
        Ingredient {
            name,
            unit,
            quantity,
            description,
            id,
        }
    }
    pub fn id(&self) -> u64 {
        self.id
    }

    pub fn set_id(&mut self, id: u64) {
        self.id = id
    }
}
